components/dashboard/src/settings/PersonalAccessTokens.tsx

```diff
@@ -452,33 +606,85 @@ function ListAccessTokensView() {
-                        No Personal Access Tokens (PAT)
-                    </h3>
-                    <p className="text-center pb-6 text-gray-500 text-base w-96">
-                        Generate a personal access token (PAT) for applications that need access to the Gitpod API.{" "}
-                    </p>
-                    <Link to={settingsPathPersonalAccessTokenCreate}>
-                        <button>New Personal Access Token</button>
-                    </Link>
-                </div>
+            {loading ? (
+                <SpinnerLoader content="fetch data" />
             ) : (
                 <>
-                    <div className="px-6 py-3 flex justify-between space-x-2 text-sm text-gray-400 mb-2 bg-gray-100 dark:bg-gray-800 rounded-xl">
-                        <h2 className="w-3/12">Token Name</h2>
-                        <h2 className="w-3/12">Permissions</h2>
-                        <h2 className="w-3/12">Expires</h2>
-                        <div className="w-3/12"></div>
-                    </div>
-                    {tokens.map((t: PersonalAccessToken) => {
-                        return <TokenEntry token={t} onDelete={handleDeleteToken} />;
-                    })}
+                    {tokens.length === 0 ? (
+                        <div className="bg-gray-100 dark:bg-gray-800 rounded-xl w-full py-28 flex flex-col items-center">
+                            <h3 className="text-center pb-3 text-gray-500 dark:text-gray-400">
+                                No Personal Access Tokens (PAT)
+                            </h3>
+                            <p className="text-center pb-6 text-gray-500 text-base w-96">
+                                Generate a personal access token (PAT) for applications that need access to the Gitpod
+                                API.{" "}
+                            </p>
+                            <Link to={settingsPathPersonalAccessTokenCreate}>
+                                <button>New Personal Access Token</button>
+                            </Link>
+                        </div>
+                    ) : (
+                        <>
+                            <div className="px-3 py-3 flex justify-between space-x-2 text-sm text-gray-400 mb-2 bg-gray-100 dark:bg-gray-800 rounded-xl">
+                                <h2 className="w-4/12">Token Name</h2>
+                                <h2 className="w-4/12">Permissions</h2>
+                                <h2 className="w-3/12">Expires</h2>
+                                <div className="w-1/12"></div>
+                            </div>
+                            {tokens.map((t: PersonalAccessToken) => (
+                                <TokenEntry
+                                    key={t.id}
+                                    token={t}
+                                    menuEntries={[
+                                        {
+                                            title: "Edit",
```

Comment:
**issue**: Not the best line to comment this, but when you _Edit_ a token without any changes to the token name or permissions, users get the following error. Shall we suppress this? ❗

<img width="512" alt="Screenshot 2022-11-30 at 2 23 17 PM" src="https://user-images.githubusercontent.com/120486/204795477-65b467fd-e011-4a18-ae0d-a9d2c5f30b3a.png">

Reply:
That error is coming from the backend. We should not suppress on the client, but we should fix the backend for this case.

Reply:
Makes sense! Also, by suppress I was suggesting to simply allowing token updates without any changes.